tests/Service/CollapsedBroadcastsService/FindByCategoryAndStartAtDateRangeTest.php

```diff
@@ -2,36 +2,98 @@
 
 namespace Tests\BBC\ProgrammesPagesService\Service\CollapsedBroadcastsService;
 
+use BBC\ProgrammesPagesService\Domain\Entity\CollapsedBroadcast;
+use BBC\ProgrammesPagesService\Domain\Entity\Genre;
 use DateTimeImmutable;
 
 class FindByCategoryAndStartAtDateRangeTest extends AbstractCollapsedBroadcastServiceTest
 {
-    public function testFindByCategoryAndStartAtDateRangeTest()
+    /**
+     * @dataProvider paginationProvider
+     */
+    public function testRepositoryReceivesCorrectParams($expectedLimit, $expectedOffset, array $paginationParams)
     {
-        $ancestry = [3];
         $fromDate = new DateTimeImmutable();
         $toDate = new DateTimeImmutable();
 
-        $category = $this->mockEntity('Genre', 3);
-        $category->method('getDbAncestryIds')->willReturn($ancestry);
+        $category = $this->createConfiguredMock(Genre::class, ['getDbAncestryIds' => [3]]);
 
-        $broadcastData = [['areWebcasts' => ['0'], 'serviceIds' => ['a', 'b']]];
-        $serviceData   = [
-            'a' => ['id' => 'bbc_one'],
-            'b' => ['id' => 'bbc_one_hd'],
+        $this->mockRepository->expects($this->once())
+             ->method('findByCategoryAncestryAndStartAtDateRange')
+             ->with($category->getDbAncestryIds(), false, $fromDate, $toDate, $expectedLimit, $expectedOffset);
+
+        $this->service()->findByCategoryAndStartAtDateRange($category, $fromDate, $toDate, ...$paginationParams);
+    }
+
+    public function paginationProvider(): array
+    {
+        return [
+            // [expectedLimit, expectedOffset, [limit, page]]
+            'CASE: default pagination' => [300, 0, []],
+            'CASE: custom pagination' => [3, 12, [3, 5]],
         ];
+    }
 
-        $this->mockRepository->expects($this->once())
+    public function testWebcastIsStripped()
+    {
+        $this->mockRepository
             ->method('findByCategoryAncestryAndStartAtDateRange')
-            ->with($ancestry, false, $fromDate, $toDate)
-            ->willReturn($broadcastData);
+            ->willReturn([
+                 ['areWebcasts' => [false, false], 'serviceIds' => [111, 222], 'broadcastIds' => [1, 2, 3, 4]],
+                 ['areWebcasts' => [true, true], 'serviceIds' => [333, 444], 'broadcastIds' => [3, 4, 56, 67]],
+                 ['areWebcasts' => [true, false], 'serviceIds' => [555, 666], 'broadcastIds' => [5, 6, 100]],
+                 ['areWebcasts' => [false, false], 'serviceIds' => [false, false], 'broadcastIds' => [7, 8, 20, 48, 23]],
+                 ['areWebcasts' => [false, false], 'serviceIds' => [true, true], 'broadcastIds' => [8, 9, 12, 122]],
+                 ['areWebcasts' => [false, false], 'serviceIds' => [null, null], 'broadcastIds' => [10, 11]],
+                 ['areWebcasts' => ['0', 0], 'serviceIds' => [777, 888], 'broadcastIds' => [14, 15]],
```

Comment:
Whats with these return values? Can the findByCategoryAncestryAndStartAtDateRange ever return data where areWebcasts is a string or int?